codecs/avif/enc/avif_enc.cpp

```diff
@@ -72,7 +76,9 @@ val encode(std::string buffer, int width, int height, AvifOptions options) {
   return val(typed_memory_view(output.size, output.data));
 }
 
-void free_result() { avifRWDataFree(&output); }
+void free_result() {
```

Comment:
Oh, I see there is `free_result` left. Are you planning to updating the memory management model in a separate PR or should it be done as part of this one?

Reply:
Will fix in this one.